*: refactor out data structures in a separate package

alaypatel07 opened the pull request:
This helps in using the same packages across different projects,
so whenever we make a change it does not need to be copy-pasted across.

Also fixes a bug in observationconfig, the setBootstrapMember was returning
a list of members and it was later getting added to etcdURLs making it a list of 
lists. The lookup on bootstrap member was failing because of this giving error, 
`member name does not exists`

alaypatel07 committed:
*: refactor out data structures in a separate package

This help in using the same packages across different projects,
so whenever we make a change it does not need to be copy pasted across

alaypatel07 commented:
@hexfusion this will probably disrupt some of your workflows, but probably better in long term

alaypatel07 committed:
configobservation: fix setBootstrapMember to return single member

The bootstrap member was return a list of members and it was later added
to etcdURLs making it a list of lists. The lookup on bootstrap was
 failing because of this giving error, `member name does not exists`.

hexfusion commented on pkg/operator/ceoutils/scaling.go:
```diff
@@ -0,0 +1,64 @@
+package ceoutils
+
```
what about just utils? ceo is kind of redundant. what do you think?

hexfusion reviewed (commented):
I think the idea is good, just a question on naming.

alaypatel07 commented:
I had it utils initially but utils.EtcdScaling did not have a good ring so changed it go ceoutils. 

I'm indifferent either way tbh, lmk if it needs a refactor

hexfusion commented:
> I had it utils initially but utils.EtcdScaling did not have a good ring so changed it go ceoutils.
> 
> I'm indifferent either way tbh, lmk if it needs a refactor

this kind of feels more like operator/api then util besides the helper function. I think we need a generic util but maybe not for these bits?

alaypatel07 commented:
+1, operator/api is a good idea, I'll refactor. Thanks

alaypatel07 committed:
*: rebase and refactor out data structures in a separate package

This help in using the same packages across different projects,
so whenever we make a change it does not need to be copy pasted across

alaypatel07 commented:
@hexfusion on second thoughts the kube convention is to use `api` for CRDs. Since we don't explicitly use the crds, its apt to use `operator/api` package name IMO. Because there can be many `apis` imported, I have explicitly imported it as `ceoapi` in order to avoid confusion.

alaypatel07 committed:
*: rebase and refactor out data structures in a separate package

This help in using the same packages across different projects,
so whenever we make a change it does not need to be copy pasted across

hexfusion reviewed (commented):
:+1:

hexfusion commented:
Thanks again @alaypatel07
